### ISSUE TITLE
Dangerous Warning Issue for React16 and componentWillReceiveProps

### ISSUE THREAD

**Issue by liliangiraudo5** [JodusNodus/react-qr-reader#204]
Hi Guys, I like your library a lot but the only problem is the appearance of a warning concerning the version of React. Indeed, if we want to switch to React 17, unfortunately the Reader component will not be updated.
Can you do the necessary? Thank you very much!

react-dom.development.js:88 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Reader

**Comment by viraljetani** [comment]
@liliangiraudo5 Yeah, I am getting the same issue when tried installing this package.
